### ISSUE TITLE
backcalc_opts

### ISSUE THREAD

**Issue by seabbs** [epiforecasts/EpiNow2#355]
This is currently quite a misleading name and I think should be changed. One potential option is to call it dencolve_opts but could also call it something linking to its underlying model more closely? 

It might also make sense to split the models into their own functions instead of the current approach with one perhaps being deconvolve and the other Rt? I’m not totally sure what the best names would be there. It would also make things tricky with higher level functions where they could be used interchangeably but not impossible (this is perhaps a larger change that should be saved for # if that goes ahead).

**Comment by sbfnk** [comment]
> This is currently quite a misleading name and I think should be changed. One potential option is to call it dencolve_opts but could also call it something linking to its underlying model more closely? As this release is now looking like it will make quite a few breaking changes it would be nice to resolve this now.
> 
> It might also make sense to split the models into their own functions instead of the current approach with one perhaps being deconvolve and the other Rt? I’m not totally sure what the best names would be there. It would also make things tricky with higher level functions where they could be used interchangeably but not impossible (this is perhaps a larger change that should be saved for #352 if that goes ahead).

That happened #345 (see [here](https://github.com/epiforecasts/EpiNow2/blob/40709d2f5351e4d8a807e04bde0f6c252db5797c/inst/stan/estimate_infections.stan#L108)) but that branch needs to be rebased to include the latest change and then the R option interface updated and finished).

**Comment by seabbs** [comment]
Aha one (or several) steps ahead. I'd forgotten how nice this was.

**Comment by seabbs** [comment]
Do you have thoughts about when we should try and add this in (i.e this release or next)? 

Along with your suggestion for dist_spec there are a lot of breaking changes and looking over the process PR it feels like there is still a fair bit to do? 

Part of me thinks we release what is on develop as is and then have a major release with all these breaking changes on. Ages weeks (basically bump up to a 2.0 release).

**Comment by sbfnk** [comment]
Yes that sounds good to me - so 1.3.3 is what's currently in `main` (not on CRAN), 1.3.4 what's currently in `develop` (to go to CRAN), and 1.4 or 2.0 will be with an updated user interface for distributions and process (but continuing with `generation_interval_opts` etc. so we're not intending to introduce them in one and then remove them in the next version?

**Comment by seabbs** [comment]
No, I think we would have to introduce and then remove in the updated version (or perhaps remove as that is still up in the air). The only thing that would be new though is `generation_time_opts()` but given that it isn't compulsory to use it doesn't seem like the end of the world. We could even update the documentation to not show its usage.

**Comment by seabbs** [comment]
`main` currently has a lot of fixes not in `1.3.3` (as released on GitHub) so is really a partial `1.3.4` (hence having an update version).
